app/controllers/confirmations_controller.rb

```diff
@@ -0,0 +1,17 @@
+class ConfirmationsController < ApplicationController
+  def verify_email
+    return unless params[:id] && params[:token]
+
+    @user = User.find_by(id: params[:id])
+
+    debugger
+    if @user.verified?
+      redirect_to login_url, notice: t(".already_verified")
+    elsif @user.auth_token_valid?(params[:token])
+      @user.update(verified_at: Time.now)
+      redirect_to login_url, notice: t(".verified")
+    else
+      redirect_to login_url, alert: t(".unsuccessful_verification")
```

Comment:
all of the conditions are redirecting to login_url and just message is decided in if else.

Remove redirect from if else and write at the end.